xdsl/dialects/linalg.py

```diff
@@ -1304,6 +1311,14 @@ def __init__(
         dimensions: Attribute,
         result: Attribute | None = None,
```

Comment:
I think I'd prefer that the `result` argument wasn't there, but I'd be happy to leave that for this PR, feel I've done enough nitting.